### PR TITLE
Add route to destroy a batch of stories

### PR DESCRIPTION
**Pull request by brunorafa-el**

### CODE REVIEW

File: id [hunk 1]
```diff
@@ -0,0 +1,1294 @@
+=> [32m#<ActiveRecord::ConnectionAdapters::PostgreSQLAdapter:0x005636568adbc0[0m
```
Comment (matheusazzi) [inline id]: Esse arquivo deve ter sido commitado sem querer @brunorafa-el

File: app/controllers/stories_bulk_destroy_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+class StoriesBulkDestroyController < ApplicationController
+  def create
+    authorize stories
+    stories.map do |story|
+      StoryOperations::Destroy.call(story, current_user)
+    end
+
+    redirect_to project, notice: "Stories was successfully destroyed."
```
Comment (matheusazzi) [inline app/controllers/stories_bulk_destroy_controller.rb]: Single quotes na notice

File: app/controllers/stories_bulk_destroy_controller.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+class StoriesBulkDestroyController < ApplicationController
+  def create
+    authorize stories
+    stories.map do |story|
```
Comment (tiarly) [inline app/controllers/stories_bulk_destroy_controller.rb]: no need for map here, maybe just `each`

File: app/controllers/stories_bulk_destroy_controller.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+class StoriesBulkDestroyController < ApplicationController
+  def create
```
Comment (JuarezLustosa) [inline app/controllers/stories_bulk_destroy_controller.rb]: why don't use delete verb ?

Comment (brunorafa-el) [reply app/controllers/stories_bulk_destroy_controller.rb]: Because delete verb doesn't have body to pass the params

File: app/controllers/stories_bulk_destroy_controller.rb (hunk 2 above)
Comment (tiarly) [inline app/controllers/stories_bulk_destroy_controller.rb]: I wonder if we need a redirect here, this controller is probably going to be used in a ajax request, maybe HTTP 200 or HTTP 204?
Also we need to respond correctly when this operation fail right, so redirect wont be acceptable here

File: app/controllers/stories_bulk_destroy_controller.rb (hunk 4 above)
Comment (tiarly) [reply app/controllers/stories_bulk_destroy_controller.rb]: Agreed, delete it's just a get request behind the scenes, it would be weird, to have an url `/stories_bulk_destroy?ids=1,2,3,4,5` and then having to split that string param, kind of a hack for something that should have been a post instead. Hence the controller name being a Destroy creation

Comment (JuarezLustosa) [reply app/controllers/stories_bulk_destroy_controller.rb]: Understood, Sorry I didn't see the params

File: spec/controllers/stories_bulk_destroy_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        project = create(:project, users: [user], teams: [user.teams.first])
+        story_1 = create(:story, project: project, requested_by: user)
+        story_2 = create(:story, project: project, requested_by: user)
+        story_3 = create(:story, project: project, requested_by: user)
+
+        sign_in user
+
+        post :create, project_id: project.id, story_ids: [story_1.id, story_2.id]
+
+        expect(project.stories).to eq([story_3])
```
Comment (tiarly) [inline spec/controllers/stories_bulk_destroy_controller_spec.rb]: If we ended up adding different responses depending on the result of this operation, we may also have to test both cases with expectations for the http response as well at a minimal i guess

File: spec/controllers/stories_bulk_destroy_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    context 'when receive an array of story ids' do
+      before(:each) do
+        post :create, project_id: project.id, story_ids: [story_1.id, story_2.id]
+      end
+
+      it 'destroys stories' do
+        expect(project.stories).to eq([story_3])
+      end
+
+      it 'responses 200' do
```
Comment (tiarly) [inline spec/controllers/stories_bulk_destroy_controller_spec.rb]: `it responds with 200`

File: spec/controllers/stories_bulk_destroy_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        expect(response).to have_http_status(200)
+      end
+
+      it 'returns a successfully message' do
+        expect(response_json('message')).to eql('Stories were successfully destroyed.')
+      end
+    end
+
+    context 'when do not destroy' do
```
Comment (tiarly) [inline spec/controllers/stories_bulk_destroy_controller_spec.rb]: `when bulk destroy fails`

File: spec/controllers/stories_bulk_destroy_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+      before(:each) do
+        allow(StoryOperations::DestroyAll).to receive(:call).and_return(false)
+        post :create, project_id: project.id, story_ids: [story_1.id, story_2.id]
+      end
+
+      it 'responses 422' do
```
Comment (tiarly) [inline spec/controllers/stories_bulk_destroy_controller_spec.rb]: `responds with 422`

File: app/controllers/stories_bulk_destroy_controller.rb [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    authorize stories
+    destroyed_stories = StoryOperations::DestroyAll.call(stories, current_user)
+
+    if destroyed_stories
+      render json: { message: 'Stories were successfully destroyed.' }, status: :ok
```
[the remark this answers is not in the document]
Comment (brunorafa-el) [reply app/controllers/stories_bulk_destroy_controller.rb]: Sure, thank you!

File: config/locales/en.yml [hunk 10]
```diff
@@ -69,6 +69,8 @@ en:
   reset_tour: "Reset tour"
   reset_tour_success: "Tour was successfully reset"
   reset_tour_fail: "An error has ocurred, please try again"
+  stories_destroy_success: "Stories were successfully destroyed"
+  stories_destroy_fail: "Stories were not successfully destroyed"
```
Comment (andrerpbts) [inline config/locales/en.yml]: I think it's better, the failure messages, be expressed in another way, without 'successfully' word in the sentence. Something like "Stories couldn't be destroyed due some validation errors"

File: app/operations/base_operations/activity_recording.rb [hunk 11]
```diff
@@ -1,23 +1,25 @@
 module BaseOperations
   module ActivityRecording
-    def fetch_project
-      case model
+    def fetch_project(record)
```
Comment (andrerpbts) [inline app/operations/base_operations/activity_recording.rb]: Why this change?

File: spec/controllers/stories_bulk_destroy_controller_spec.rb [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+    let(:story_2) { create(:story, project: project, requested_by: user) }
+    let(:story_3) { create(:story, project: project, requested_by: user) }
+
+    context 'when receive an array of story ids' do
+      before(:each) do
+        post :create, project_id: project.id, story_ids: [story_1.id, story_2.id]
+      end
+
+      it 'destroys stories' do
+        expect(project.stories).to eq([story_3])
```
Comment (adbatista) [inline spec/controllers/stories_bulk_destroy_controller_spec.rb]: expect ... to contain_exactly(story_3)

File: spec/controllers/stories_bulk_destroy_controller_spec.rb [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+        expect(response).to have_http_status(200)
+      end
+
+      it 'returns a successfully message' do
+        expect(response_json('message')).to eql('Stories were successfully destroyed')
+      end
+    end
+
+    context 'when bulk destroy fails' do
+      before(:each) do
```
Comment (adbatista) [inline spec/controllers/stories_bulk_destroy_controller_spec.rb]: (:each) is unecessary, avoid to use this

File: app/operations/base_operations/activity_recording.rb (hunk 11 above)
Comment (brunorafa-el) [reply app/operations/base_operations/activity_recording.rb]: Because I'm receiving a collection of stories over the model, so I iterated with the model in line 15 passing the record as a story!

File: spec/controllers/stories_bulk_destroy_controller_spec.rb (hunk 12 above)
Comment (brunorafa-el) [reply spec/controllers/stories_bulk_destroy_controller_spec.rb]: Thank you!

File: spec/controllers/stories_bulk_destroy_controller_spec.rb (hunk 13 above)
Comment (brunorafa-el) [reply spec/controllers/stories_bulk_destroy_controller_spec.rb]: Thank you!

File: config/locales/en.yml (hunk 10 above)
Comment (brunorafa-el) [reply config/locales/en.yml]: Of course looks very weird having "successfully" in a failure message, I don't know how I don't think in it. Thanks!

File: app/operations/base_operations.rb [hunk 14]
```diff
@@ -63,4 +63,16 @@ def operate!
...
+
+  class DestroyAll < BaseOperations::Create
+    def self.name
+      'destroy'
+    end
+
+    protected
+
+    def operate!
+      model.destroy_all
```
Comment (andrerpbts) [inline app/operations/base_operations.rb]: Shouldn't this runing inside a transaction?

File: CHANGELOG.md [hunk 15]
```diff
@@ -2,6 +2,10 @@
 
 ## [Unreleased]
 
+## [1.19.0] 2018-03-06
```
Comment (andrerpbts) [inline CHANGELOG.md]: Remove this tag, a new tag will be generated when releasing.
